compiler warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size 3012

```
[90/182] Building C object src/nvim/CMakeFiles/nvim.dir/log.c.o
../src/nvim/log.c: In function ‘log_callstack_to_file’:
../src/nvim/log.c:202:50: warning: ‘%s’ directive output may be truncated writing up to 4095 bytes into a region of size 3012 [-Wformat-truncation=]
   snprintf(cmdbuf, sizeof(cmdbuf), "addr2line -e %s -f -p", exepath);
                                                  ^~         ~~~~~~~
../src/nvim/log.c:202:3: note: ‘snprintf’ output between 20 and 4115 bytes into a destination of size 3025
   snprintf(cmdbuf, sizeof(cmdbuf), "addr2line -e %s -f -p", exepath);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The code appears to be older already (eb2473e9e, from @justinmk).

gcc (GCC) 8.1.1 20180531

--- a/src/nvim/log.c
+++ b/src/nvim/log.c
@@ -195,7 +195,7 @@ void log_callstack_to_file(FILE *log_file, const char *const func_name,
   }
   assert(24 + exepathlen < IOSIZE);  // Must fit in `cmdbuf` below.
 
-  char cmdbuf[IOSIZE + (20 * ARRAY_SIZE(trace))];
+  char cmdbuf[IOSIZE + (20 * ARRAY_SIZE(trace)) + MAXPATHL];
   snprintf(cmdbuf, sizeof(cmdbuf), "addr2line -e %s -f -p", exepath);
   for (int i = 1; i < trace_size; i++) {
     char buf[20];  // 64-bit pointer 0xNNNNNNNNNNNNNNNN with leading space.

